modules/Microsoft.App/managedEnvironments/deploy.bicep

```diff
@@ -0,0 +1,197 @@
+@description('Required. Name of the Container Apps Managed Environment.')
+param name string
+
+@description('Optional. Existing Log Analytics Workspace resource ID.')
+param logAnalyticsWorkspaceResourceId string = ''
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Tags of the resource.')
+param tags object = {}
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalId\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments array = []
+
+@allowed([
+  'Consumption'
+  'Premium'
+])
+@description('Optional. Managed environment SKU.')
+param skuName string = 'Consumption'
+
+@description('Optional. Logs destination.')
+param logsDestination string = 'log-analytics'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool
+
+@description('Optional. Application Insights connection string used by Dapr to export Service to Service communication telemetry.')
+@secure()
+param daprAIConnectionString string = ''
+
+@description('Optional. Azure Monitor instrumentation key used by Dapr to export Service to Service communication telemetry.')
+@secure()
+param daprAIInstrumentationKey string = ''
+
+@description('Optional. CIDR notation IP range assigned to the Docker bridge, network. Must not overlap with any other provided IP ranges.')
+param dockerBridgeCidr string = ''
+
+@description('Optional. Resource ID of a subnet for infrastructure components. This subnet must be in the same VNET as the subnet defined in runtimeSubnetId. Must not overlap with any other provided IP ranges.')
+param infrastructureSubnetId string = ''
+
+@description('Optional. Boolean indicating the environment only has an internal load balancer. These environments do not have a public static IP resource. They must provide runtimeSubnetId and infrastructureSubnetId if enabling this property.')
+param internal bool = false
+
+@description('Optional. Configuration used to control the Environment Egress outbound traffic.')
```

Comment:
Testing this parameter is also not working :(

```powershell
   - The environment network configuration is invalid: OutboundSettings is not supported. (Code:ManagedEnvironmentInvalidNetworkConfiguration)
```
Even when trying different settings.

It also doesn't allow me to inject it as a parameter with conditions similar to the log analytics issue

```powershell
 - Validation failed for a resource. Check 'Error.Details[0]' for more information. (Code: ValidationForResourceFailed)
   - Invalid request body for environment. Path: $.vnetConfiguration.outboundSettings.outBoundType. Does not conform to Managed Environment schema, please visit for more information https://docs.microsoft.com/azure/container-apps/azure-resource-manager-api-spec?tabs=arm-template#container-apps-environment (Code:ManagedEnvironmentInvalidSchema)
 ```

Only works when doing something like the following directly in the bicep deploy :

```bicep
      outboundSettings: {
        outBoundType: 'UserDefinedRouting'
        virtualNetworkApplianceIp: '10.10.10.10'
      }
```

Reply:
Removed